File: Prime-generator/Javascript/2.js
function is_prime(n) {
    for (let i=2; i<Math.sqrt(n); i++) {
        if (n % i == 0)
            return false
    }
    return true
}



const start = process.hrtime.bigint();
const primes = [];
for (let i=2; i<process.argv[2]; i++) {
    if (is_prime(i)) {
        primes.push(i)
    }
}
const end = process.hrtime.bigint();
const duration = end - start;
console.log(`Execution time: ${duration} ns`);

File: Prime-generator/Python/2.py
import sys
import time
from math import sqrt


def is_prime(n):
    for i in range(2, int(sqrt(n))):
        if n % i == 0:
            return False
    return True



start = time.time_ns()
primes = []
for i in range(2, int(sys.argv[1])):
    if is_prime(i):
        primes.append(i)
duration = time.time_ns() - start
print(f"Execution time: {duration} ns")
